src/components/MeasurementsToolList.vue

```diff
@@ -0,0 +1,94 @@
+<script setup lang="ts">
+import { computed } from 'vue';
+import { StoreState, StoreActions } from 'pinia';
+import { useCurrentImage } from '../composables/useCurrentImage';
+import { useAnnotationTool } from '../store/tools/useAnnotationTool';
+import { AnnotationTool } from '../types/annotation-tool';
+
+type ToolFactory = (...args: any) => AnnotationTool;
+type UseAnnotationTool = ReturnType<
+  typeof useAnnotationTool<ToolFactory, unknown>
+>;
+type AnnotationToolStore = StoreState<UseAnnotationTool> &
+  StoreActions<UseAnnotationTool>;
+
+const props = defineProps<{
+  toolStore: AnnotationToolStore;
```

Comment:
How to type `toolStore` so don't have to use any in parent components?

MeasurementsRulerList.vue:
```
<MeasurementsToolList :toolStore="toolStore as any" icon="mdi-ruler">
```

Reply:
Fundamentally, there's a type mismatch between `AnnotationToolStore` and the actual tool stores where we can't assign a tool store to the type `AnnotationToolStore`. We could add generics to `AnnotationToolStore` to make up for (hopefully all) of it.

We could also define a smaller interface for what is necessary in MeasurementsToolList.
```
interface ToolStore {
  toolByID: Record<string, AnnotationTool>;
  removeTool<T extends string>(id: T): void;
  ...
}
const props = defineProps<{
  toolStore: ToolStore,
  ...
```

Reply:
Good tips!  So fun... and frustrating.  

Last commit makes `id` generic.  (I just `Omit`ed the serialize stuff.)  Interested in your thoughts.

Reply:
+1. I like keeping `serialize` and `deserialize` out of the AnnotationTool type.